### PR TITLE
Diffusion PA kernel fix [diffusion_pa_ker_revert]

### PR DESCRIPTION
**Pull request by camierjs**
Remove the indices computation for `SmemPADiffusionApply3D` that added extra requirements on `dofs1D` and `quad1D`.
The instanciatied kernels with `dofs1D==quad1D` are now also tested in the unit tests.

 | PR | Author | Editor | Reviewers | Assignment | Approval | Merge|
 | --- | --- | --- | --- | --- | --- | --- |
| [#2861](https://github.com/mfem/mfem/pull/2861) | @camierjs | @tzanio | @tzanio + @v-dobrev | 3/1/22 | 3/2/22 | 3/3/22 | |

### CODE REVIEW

**Comment by tzanio** [comment]
This PR is now [under review](https://github.com/mfem/mfem/blob/master/CONTRIBUTING.md#pull-requests) (see the table in the PR description). To help with the review process, please **do not force push** to the branch.

File: fem/bilininteg_diffusion_pa.cpp [hunk 1]
```diff
@@ -1601,12 +1574,8 @@ static void SmemPADiffusionApply3D(const int NE,
...
             MFEM_FOREACH_THREAD(qx,x,Q1D)
             {
-               const int i = qi(qx,dy,Q1D);
-               const int j = dj(qx,dy,D1D);
-               const int k = qk(qx,dy,Q1D);
-               const int l = dl(qx,dy,D1D);
-               B[i][j] = b(qx,dy);
-               G[k][l] = g(qx,dy) * sign(qx,dy);
+               B[qx][dy] = b(qx,dy);
+               G[qx][dy] = g(qx,dy);
```
Comment (YohannDudouit) [inline fem/bilininteg_diffusion_pa.cpp:1578]: I think it's better to remove the `x` and `y`:
```suggestion
               B[q][d] = b(q,d);
               G[q][d] = g(q,d);
```

File: fem/bilininteg_diffusion_pa.cpp [hunk 2]
```diff
@@ -1702,16 +1656,12 @@ static void SmemPADiffusionApply3D(const int NE,
...
+            MFEM_FOREACH_THREAD(qx,x,Q1D)
             {
-               const int i = qi(q,d,Q1D);
-               const int j = dj(q,d,D1D);
-               const int k = qk(q,d,Q1D);
-               const int l = dl(q,d,D1D);
-               Bt[j][i] = b(q,d);
-               Gt[l][k] = g(q,d) * sign(q,d);
+               Bt[dy][qx] = b(qx,dy);
+               Gt[dy][qx] = g(qx,dy);
```
Comment (YohannDudouit) [inline fem/bilininteg_diffusion_pa.cpp:1664]:
```suggestion
               Bt[d][q] = b(q,d);
               Gt[d][q] = g(q,d);
```

File: fem/bilininteg_diffusion_pa.cpp (hunk 1 above)
Comment (camierjs) [reply fem/bilininteg_diffusion_pa.cpp:1578]: `d` is already used as local variable name and it is a way to be consistent with the `MFEM_FOREACH_THREAD` arguments.

**Review by tzanio — APPROVED** [review]

**Review by YohannDudouit — APPROVED** [review]

**Comment by tzanio** [comment]
Merged in `next` for testing...

**Pull request merged**